### DISCUSSION TITLE
Vscode failed to open firefox for github authentication if firefox is already opened.

### DISCUSSION THREAD

**Discussion by harryhanYuhao** [i3/i3#5705] [Q&A]
## I'm submitting a…

<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

When firefox is already opened, it is impossible to authenticate vscode with github account. When the request for authentication is sent through vscode (by clicking `sign in to sync setting` button)
apparently vscode try to open a tab in firefox, which is not possible if firefox is already opened. The following error message will be shown: `Firefox is already running, but is not responding. To use firefox, you must first close the existing firefox process, restart your device, or use a different profile` (see the attached screenshot). Note I have firefox running normally in workspace 2.

![i3_bug](https://github.com/i3/i3/assets/116114033/f01fc819-39af-4c7a-b9e4-a40ce3ef6596)
 

## Expected Behavior

Vscode can successfully authenticate by opening a new firefox tab.

## Reproduction Instructions

- open firefox
- sign in through vscode by clicking `sign in to sync setting` button

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.22 (2023-01-02) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.22 (2023-01-02) (pid 1900)
Loaded i3 config:
  /home/hhyh/.config/i3/config (main) (last modified: Sat 30 Sep 2023 13:45:39 BST, 795 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

The problem exists with default config
<details><summary>Config file</summary><pre>
</pre>
</details>

<pre>
Logfile URL:
</pre>

<pre>
- Linux Distribution & Version: Fedora 38 on x86_64,
- Are you using a compositor (e.g., xcompmgr or compton): no
</pre>

**Comment by orestisfl** [comment]
I converted this to a discussion because I don't really see how i3 is involved in this. You can also try to reproduce this in a different WM like openbox to see if that has any effect on your issue.

I would also try to run firefox from the cli to see if there is any error message when you try to open it from vscode.
